raster/src/main/scala/geotrellis/raster/merge/SinglebandTileMergeMethods.scala

```diff
@@ -154,4 +154,69 @@ trait SinglebandTileMergeMethods extends TileMergeMethods[Tile] {
       case _ =>
         self
     }
+
+  def union(extent: Extent, otherExtent: Extent, other: Tile, method: ResampleMethod, unionF: (Option[Double], Option[Double]) => Double): Tile = {
+    val unionInt = (l: Option[Double], r: Option[Double]) => unionF(l, r).toInt
```

Comment:
Hm, mb the signature `(l: Option[Int], r: Option[Int]) => Int` is a bit more clear, any way expensive casts occur. I'm actually wondering what's the cost of those.